### PR TITLE
Linux Audio feature added

### PR DESCRIPTION
**Pull request by bhaveshAn**
Linux audio feature is added and it doesn't having any dependencies. Instead, feature is running using gstreamer only.

Please view this PR.

### CODE REVIEW

File: plyer/platforms/linux/audio.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    def __init__(self, file_path=None):
+        default_path = '/home/recording.wav'
+        super(LinuxAudio, self).__init__(file_path or default_path)
+
+    def _start(self):
+        process = subprocess.Popen(("gst-launch", "autoaudiosrc",
+                                    "num-buffers=100", "!", "audioconvert",
+                                    "!", "vorbisenc", "!", "oggmux", "!",
+                                    "filesink", "location=" + self.file_path),
+                                   stdout=subprocess.PIPE)
```
Comment (malverick) [inline plyer/platforms/linux/audio.py]: add an extra parameter 'shell=True' to the `Popen` call.

File: plyer/platforms/linux/audio.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import subprocess
+
+
+class LinuxAudio(Audio):
+    def __init__(self, file_path=None):
+        default_path = '/home/recording.wav'
+        super(LinuxAudio, self).__init__(file_path or default_path)
+
+    def _start(self):
+        process = subprocess.Popen(("gst-launch", "autoaudiosrc",
```
Comment (malverick) [inline plyer/platforms/linux/audio.py]: should be `self.process`

File: plyer/platforms/linux/audio.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                                    "num-buffers=100", "!", "audioconvert",
+                                    "!", "vorbisenc", "!", "oggmux", "!",
+                                    "filesink", "location=" + self.file_path),
+                                   stdout=subprocess.PIPE)
+        process.wait()
```
Comment (malverick) [inline plyer/platforms/linux/audio.py]: `self.process.wait()`

File: plyer/platforms/linux/audio.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+    def _stop(self):
+        process.terminate()
```
Comment (malverick) [inline plyer/platforms/linux/audio.py]: `self.process.terminate()`

**Comment by malverick** [comment]
@bhaveshAn initially when I compiled your code, logs generated were:
[initial logs](https://gist.github.com/malverick/6bcc14c63b5bb1b11c94e8098c955c4a)
After doing the above changes, logs are - 
[logs](https://gist.github.com/malverick/c339836c7a18ac8238d758bea978c1f6)

As I press the `Start Recording` button, I get `autoaudiosrc: 1: autoaudiosrc: gst-launch: not found` on terminal.
After I press the `Press to Stop Recording` button, I get ` OSError: [Errno 3] No such process` on terminal.

**Comment by bhaveshAn** [comment]
@malverick Please try it using command line as well which is provided in facade.

**Comment by bhaveshAn** [comment]
Tested and working fine on all versions of Linux Ubuntu.
However, this has not been merged so I am providing the link to [illustration of the feature](https://www.youtube.com/watch?v=dp4T5t-9mS8)
It does not having any dependency either. Hence in a merge-able state

File: plyer/facades/temperature.py [hunk 5]
```diff
@@ -1,5 +1,6 @@
 # coding=utf-8
 
+
```
Comment (KeyWeeUsr) [inline plyer/facades/temperature.py]: Same as previous.

File: plyer/platforms/linux/audio.py [hunk 6]
```diff
@@ -0,0 +1,27 @@
+from plyer.facades.audio import Audio
+import subprocess
+
+
+class LinuxAudio(Audio):
```
Comment (KeyWeeUsr) [inline plyer/platforms/linux/audio.py]: This is definitely a GStreamer used as a provider for audio, not a "Linux" audio, so it'd nice to name it appropriately. There are other providers and I believe gnome or [kde](https://en.wikipedia.org/wiki/Amarok_(software)) has even its own minimalistic one.

**Comment by akshayaurora** [comment]
this makes streamer a dependency...

**Pull request closed without merging**